sync/awsdatabricks.py

```diff
@@ -353,28 +207,7 @@ def _get_cluster_report(
     )
 
 
-def _get_cluster_instances_from_s3(bucket: str, file_key: str, cluster_id):
-    s3 = boto.client("s3")
-    try:
-        return s3.get_object(Bucket=bucket, Key=file_key)["Body"].read()
-    except ClientError as ex:
-        if ex.response["Error"]["Code"] == "NoSuchKey":
-            logger.warning(
-                f"Could not find sync_data/cluster_instances.json for cluster: {cluster_id}"
-            )
-        else:
-            logger.error(
-                f"Unexpected error encountered while attempting to fetch sync_data/cluster_instances.json: {ex}"
-            )
-
-
-def _get_cluster_instances_from_dbfs(filepath: str):
-    filepath = format_dbfs_filepath(filepath)
-    dbx_client = get_default_client()
-    try:
-        return read_dbfs_file(filepath, dbx_client)
-    except Exception as ex:
-        logger.error(f"Unexpected error encountered while attempting to fetch {filepath}: {ex}")
+sync._databricks._get_cluster_report = _get_cluster_report
```

Comment:
Interesting. How does this work if you try to use both the Azure and AWS modules? This kind of seems like you get one or the other

Reply:
Yeah, good call-out. You can't use both modules together. I don't love the patching here, but I couldn't see a better way without duplicating a bunch of code - I should've encapsulated the API in classes rather than modules at the start. For now though, perhaps a worthwhile guard is to raise an error if someone tries to import both.

Configuration only allows 1 Databricks workspace so there's no reason to import both.

Reply:
Yeah, there are definitely some practical limitations to using both modules, so I don't think this is a "blocking" issue, but yeah, a basic guardrail might be good just in case